src/sentry/utils/performance_issues/detectors/experiments/n_plus_one_db_span_detector.py

```diff
@@ -277,3 +264,38 @@ def contains_complete_query(span: Span, is_source: bool | None = False) -> bool:
         return True
     else:
         return query and not query.endswith("...")
+
+
+def get_db_span_description(span: Span) -> str:
+    """
+    For MongoDB spans, we use the `description` provided by Relay since it re-includes the collection name.
+    See https://github.com/getsentry/relay/blob/25.3.0/relay-event-normalization/src/normalize/span/description/mod.rs#L68-L82
+    """
+    default_description = span.get("description", "")
+    db_system = span.get("sentry_tags", {}).get("system")
+    return (
+        span.get("sentry_tags", {}).get("description", default_description)
+        if db_system == "mongodb"
+        else default_description
+    )
+
+
+def are_spans_equivalent(a: Span, b: Span) -> bool:
+    """
+    Returns True if two DB spans are sufficiently similar for grouping N+1 DB Spans
+    """
+    hash_match = a.get("hash") == b.get("hash")
+    has_description = bool(a.get("description"))
+    description_match = a.get("description") == b.get("description")
+    base_checks = all([hash_match, has_description, description_match])
+
+    a_db_system = a.get("sentry_tags", {}).get("system")
+    # We perform more checks for MongoDB spans
+    if a_db_system == "mongodb":
+        # Relay augments MongoDB span descriptions with more collection data.
+        # We can use this for more accurate grouping.
+        a_relay_description = a.get("sentry_tags", {}).get("description")
+        b_relay_description = b.get("sentry_tags", {}).get("description")
+        return a_relay_description == b_relay_description and base_checks
+
+    return base_checks
```

Comment:
Now, instead of just checking `a.hash` against `b.hash` we're also validating that the descriptions match, and aren't empty. These two checks used to only happen after the detector ran through, in `_contains_valid_repeating_query` and only for the first of the `n_spans`. Now we're checking every span against the previous explicitly. 

For `mongodb` system spans we also check the relay descriptions which include collection names, for even more confidence.